server-backend/src/SwapService.ts

```diff
@@ -68,7 +68,15 @@ export class SwapService implements OnApplicationBootstrap, OnApplicationShutdow
         const paymentHash = hashTag.data;
 
         const outputAmount = this.getCheckedAmount(new Decimal(satoshis).div(1e8).toDecimalPlaces(8));
-        const timeoutBlockHeight = (await this.bitcoinService.getBlockHeight()) + this.swapConfig.lockBlockDelta.in;
+        /**
+         * The lockBlockDeltaIn parameter specifies a custom CLTV expiry (in blocks) for the swap.
+         * If not provided, the default value from the swap configuration is used.
+         */
+        const lockBlockDeltaIn = request.lockBlockDeltaIn ?? this.swapConfig.lockBlockDelta.in;
+        if (lockBlockDeltaIn < 144) {
```

Comment:
I'd put the 144 in config. You can add this to the config:
```
swapConfig: {
  minDeltaIn: z.number().positive().default(144)
```
and then read it like this:
```suggestion
        if (lockBlockDeltaIn < this.swapConfig.minDeltaIn) {
```